app/js/lib/foxbox/services/ip-camera.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+import BaseService from './base';
+
+const TYPE = 'IpCamera';
+
+const p = Object.freeze({
+  // Public getter only properties.
+  manufacturer: Symbol('manufacturer'),
+  model: Symbol('model'),
+  name: Symbol('name'),
+
+  // Private getter only properties.
+  adapter: Symbol('adapter'),
+  getters: Symbol('getters'),
+  setters: Symbol('setters'),
+});
+
+export default class IpCameraService extends BaseService {
+  constructor(props, config) {
+    super(props, config);
+
+    // Public getter only properties.
+    this[p.manufacturer] = props.properties.manufacturer;
+    this[p.model] = props.properties.model;
+    this[p.name] = props.properties.name;
+
+    // Private getter only properties.
+    this[p.adapter] = props.adapter;
+    this[p.getters] = props.getters;
+    this[p.setters] = props.setters;
+
+    Object.seal(this);
+  }
+
+  // Public getter only properties.
+  get type() {
+    return TYPE;
+  }
+
+  get manufacturer() {
+    return this[p.manufacturer];
+  }
+
+  get model() {
+    return this[p.model];
+  }
+
+  get name() {
+    return this[p.name];
+  }
+
+  // Public dynamic property getters.
+  getImageList() {
+    return this[p.query]('image_list');
+  }
+
+  getLatestImage() {
+    return this[p.query]('latest image');
+  }
+
+  // Public methods.
+  takeSnapshot() {
```

Comment:
I really like the idea of moving as much code as possible to the base class. I also think simplifying the code makes a lot of sense.

I'm not so much attached to the internals than to the publicly exposed API. If doing these changes means less effort when writing classes for new adapters, I'm supporting it!

Do you want me to implement it before you continue the review?